### ISSUE TITLE
Rust 2018 migration

### ISSUE THREAD

**Issue by SaadiSave** [SLMT/telnet-rs#18]
Is this crate planning to upgrade to Rust 2018? I can contribute a PR if needed.

**Comment by SLMT** [comment]
Sure! BTW, I am not sure what should be modified for Rust 2018, is there any reference for this?

**Comment by SaadiSave** [comment]
I have already finished. Fyi, you can use `cargo migrate` after adding `edition = 2018` to `Cargo.toml`. I have already finished migration to 2018, using a different method:

1. Set the edition in Cargo.toml
2. Follow all rust-analyzer and clippy suggestions in the Vscode.

I will be making a PR soon.

Btw, what's the copyright situation like? Do I add my name to the copyright or something? Please clarify that.

I still have to write some tests. I also have ideas for an async implementation, if you'd like to work on it with me.

**Comment by SLMT** [comment]
Ok. Sounds good!

I totally forgot that I use my name in the Copyright statement. Do you think if it would be better to replace my name with `telnet.rs Contributors`?

We can discuss the aynsc implementation after we merge the PR.

**Comment by SaadiSave** [comment]
Something like `Copyright (c) 2017-2021 Yushan Lin and telnet-rs contributors` and then maybe store the contributors list in `Contributors.md`?

Also, I observed that you have not renewed the copyright since 2017. Renewing it every year is recommended.

**Comment by SLMT** [comment]
Thanks for your advice. I will update the Copyright statement and add `Contributors.md` immediately.

**Comment by SLMT** [comment]
@SaadiSave Could you help me check if [this PR](https://github.com/SLMT/telnet-rs/pull/19) looks good for you?

**Comment by SaadiSave** [comment]
Yes, looks good. Btw, I should be done documenting my changes by Friday.

**Comment by SLMT** [comment]
Great! I'm looking forward to seeing it.

**Comment by SaadiSave** [comment]
I've added some code for error handling, so that errors are not just strings anymore. The PR should land before Sunday.

**Comment by SaadiSave** [comment]
The PR has landed
